apps/keyboard/js/settings/user_dictionary.js

```diff
@@ -52,14 +52,22 @@ UserDictionary.prototype.getList = function() {
   });
 };
 
-// Save the list of words to indexedDB. We queue the saves such that firing
-// successive saves won't easily break. Meanwhile, the queued save has to
-// resolve back to the view to update the UI accordingly.
+// Save the list of words and the dictionary blob to indexedDB.
+// We queue the saves such that firing successive saves won't easily break.
+// Meanwhile, the queued save has to resolve back to the view to update the
+// UI accordingly.
 // XXX: implement abortable queue for successive saves such that we skip
 //      not-yet-executed intermediate saves.
-UserDictionary.prototype._saveList = function() {
+// XXX: blob generation can be time-consuming. WebWorker-ize it?
+UserDictionary.prototype._saveDict = function() {
+  var wordList = Array.from(this._wordSet);
+
+  var dictBlob = new TSTConverter(wordList).toBlob();
+
   var p = this._saveQueue.then(
-    () => this._dbStore.setItem('wordlist', Array.from(this._wordSet))
+    () => this._dbStore.setItem('wordlist', wordList)
+  ).then(
+    () => this._dbStore.setItem('dictblob', dictBlob)
```

Comment:
How about implement `Promise#setItems` and use that here? I am sorry that I didn't do that back then ... I thought I did.